python/ray/train/_checkpoint.py

```diff
@@ -81,13 +81,16 @@ def _validate_checkpoint_strategy(self):
             self._checkpoint_strategy.checkpoint_score_attribute = TIMESTAMP
 
     def _load_checkpoint(
-        self, checkpoint_to_load: Optional[Union[Dict, str, Path]]
-    ) -> Optional[Dict]:
+        self, checkpoint_to_load: Optional[Union[Dict, str, Path, Checkpoint]]
+    ) -> Optional[Union[Dict, Checkpoint]]:
```

Comment:
Can we change this to only accept Checkpoint? That will help clean this up a lot. We don't need backwards compatibility here.